### DISCUSSION TITLE
Wasm doesn't work in Go 1.17

### DISCUSSION THREAD

**Discussion by ghost** [hajimehoshi/ebiten#1781] [Q&A] [answered]
I followed the steps on the https://ebiten.org/documents/webassembly.html (Option2), 
compiling my ebiten project into wasm with go 1.17 compiler, but it doesn't work.
(I also tried the 2048 game in the examples, it also failed)

The console of browser shows the following error:
```Uncaught (in promise) LinkError: WebAssembly.instantiate(): Import #5 module="go" function="runtime.walltime" error: function import requires a callable```

I rollback to the go 1.16.7 compiler and it works fine, this problem seems to be only occurs on go 1.17

compiler version: go1.17 linux/amd64
command: ```GOOS=js GOARCH=wasm go build -o project.wasm```

**Comment by hajimehoshi** [comment, marked as answer]
My guess is that you need the latest (Go 1.17's) `wasm_exec.js`.

See also:

* https://github.com/golang/go/commit/862ddb37b4e24bcb89f060d50325c8f863ee99bc
* https://github.com/golang/go/issues/47082

**Reply by ghost** [reply]
I replaced my old wasm_exec.js with the new one, and this makes the ebiten2048 works noramlly.
Thank you!
But I faced another problem about goroutines in my project, and i didn't find a similar issue or discussion about it.
Should I ask here or open a new one?

**Reply by hajimehoshi** [reply]
Please open a new discussion, thanks!
